app/Http/Controllers/UserController.php

```diff
@@ -109,15 +110,56 @@ public function store(Request $request)
         }
     }
 
-    public function createActivity()
+    public function check(Request $request)
```

Comment:
`check()` function is to long, and it is doing a lot of things. Try set getters in you Models to query the Answer, Question and Choices and create a helper to check the results. And I want you to separate the `check()` to a ResultController because if you think OOP way results is related to user but somehow not part of User.

Reply:
Always remember this point. A function should "DO ONE THING" this idea is based from famous book "The clean code" by Robert C. Martin and it highly recommended to refrain from bad code